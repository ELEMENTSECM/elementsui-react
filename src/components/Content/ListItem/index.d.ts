import * as React from 'react';

export interface ListItemProps {
	/**
	 * Any JSX
	 */
	children?: any;
	/**
	 * onClick event handler
	 */
	onClick?: (...args: any[]) => any;
	/**
	 * Item link
	 */
	link?: string;
	/**
	 * Icon class name
	 */
	icon?: string;
	/**
	 * Icon color
	 */
	iconColor?: string;
	/**
	 * List item class name
	 */
	className?: string;
	/**
	 * Link className
	 */
	linkClassName?: string;
}

declare const ListItem: React.SFC<ListItemProps>;

export default ListItem;
