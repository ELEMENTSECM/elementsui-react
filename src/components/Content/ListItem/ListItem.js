import * as React from "react";
import PropTypes from "prop-types";
import { styles } from "./ListItem.styles";
import { classNamesFunction, customizable, styled } from "office-ui-fabric-react/lib/Utilities";

export function ListItem(props) {
	const { styles, children, onClick, link } = props;
	const classNames = classNamesFunction()(styles, props);
	return (
		<li className={classNames.root}>
			<a className={classNames.link} href={link} onClick={onClick}>
				<div className={classNames.row}>
					<div className={classNames.iconCol}>
						<i className={classNames.icon} />
					</div>
					<div className={classNames.textCol}>
						<span>
							<span>{children}</span>
						</span>
					</div>
				</div>
			</a>
		</li>
	);
}

ListItem.propTypes = {
	/** Any JSX */
	children: PropTypes.any,
	/** onClick event handler */
	onClick: PropTypes.func,
	/** Item link */
	link: PropTypes.string,
	/** User-defined styling */
	styles: PropTypes.func
};

export default styled(customizable("ListItem", [ "theme" ])(ListItem), styles);
