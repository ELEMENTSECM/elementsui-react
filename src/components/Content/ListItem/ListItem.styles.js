export const styles = props => {
	const { className } = props;

	return {
		root: [
			'ListItem',
			{
				overflow: 'hidden',
				marginBottom: -1,
				position: 'relative',
				selectors: {
					':nth-child(odd)': {
						background: '#f7f7f7'
					}
				}
			},
			className
		],
		link: [
			{
				fontSize: '1.2em',
				minHeight: 49,
				padding: '15px 20px 4px 37px',
				marginLeft: 17,
				borderRadius: 0,
				borderWidth: 0,
				color: '#555',
				display: 'block'
			}
		],
		icon: [
			{
				marginLeft: -13
			}
		],
		row: ['row'],
		iconCol: ['col-xs-1'],
		textCol: ['col-xs-10']
	};
};
