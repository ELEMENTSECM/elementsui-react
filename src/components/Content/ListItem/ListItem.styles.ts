import { css } from "react-emotion";

export const styles = {
	ListItem: css`
		overflow: hidden;
		margin-bottom: -1px;
		position: relative;
		'&:nth-child(odd)': {
			background: #f7f7f7;
		}
	`,
	link: css`
		font-size: 1.2em;
		min-height: 49px;
		padding: 15px 20px 4px 37px;
		margin-left: 17px;
		border-radius: 0;
		border-width: 0;
		color: #555;
		display: block;
	`,
	icon: css`margin-left: -13px;`
};
