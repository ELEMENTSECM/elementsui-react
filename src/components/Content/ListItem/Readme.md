List item
```js
	<ul>
			{[
		{
			title: 'Notification',
			icon: 'glyphicon glyphicon-large glyphicon-bell',
			iconColor: '#3d9c97'
		},
		{
			title: 'Case',
			icon: 'glyphicon glyphicon-large glyphicon-folder-close',
			iconColor: '#428bca'
		},
		{
			title: 'Registry Entry',
			icon: 'glyphicon glyphicon-large glyphicon-envelope',
			iconColor: '#ffad5b'
		}
	].map((x, i) => (
				<ListItem key={i} icon={x.icon} iconColor={x.iconColor} onClick={() => alert(`List item ${x.title} selected`)}>
					{x.title}
				</ListItem>
			))}
	</ul>

```
