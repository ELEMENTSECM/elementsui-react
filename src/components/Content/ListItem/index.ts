export { default, IListItemProps } from "./ListItem";
