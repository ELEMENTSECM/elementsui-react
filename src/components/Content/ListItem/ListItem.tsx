import * as React from "react";
import { styles } from "./ListItem.styles";

export interface IListItemProps {
	/**
	 * Any JSX
	 */
	children?: any;
	/**
	 * onClick event handler
	 */
	onClick?: (...args: any[]) => any;
	/**
	 * Item link
	 */
	link?: string;
	/**
	 * Icon class name
	 */
	icon?: string;
	/**
	 * Icon color
	 */
	iconColor?: string;
	/**
	 * List item class name
	 */
	className?: string;
	/**
	 * Link className
	 */
	linkClassName?: string;
}

const ListItem: React.SFC<IListItemProps> = (props) => {
	const { children, onClick, link, icon, iconColor, className, linkClassName } = props;
	return (
		<li className={`${styles.ListItem} ${className}`}>
			<a className={`${styles.link} ${linkClassName}`} href={link} onClick={onClick}>
				<div className="row">
					<div className="col-xs-1">
						<i style={iconColor && { color: iconColor }} className={`${styles.icon} ${icon}`} />
					</div>
					<div className="col-xs-10">
						<span>
							<span>{children}</span>
						</span>
					</div>
				</div>
			</a>
		</li>
	);
};

export default ListItem;
