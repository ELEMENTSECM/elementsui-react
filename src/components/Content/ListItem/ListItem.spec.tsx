import * as React from "react";
import { shallow } from "enzyme";
import ListItem from "./index";

describe("ListItem", () => {
	test("displays correctly", () => {
		const wrapper = shallow(<ListItem link="#">Test title</ListItem>);
		expect(wrapper).toMatchSnapshot();
	});
});
