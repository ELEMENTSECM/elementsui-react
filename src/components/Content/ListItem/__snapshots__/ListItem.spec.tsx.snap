// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    link="#"
  >
    Test title
  </ListItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="css-19bj1hn undefined"
        href="#"
        onClick={undefined}
      >
        <div
          className="row"
        >
          <div
            className="col-xs-1"
          >
            <i
              className="css-1qbtcil undefined"
              style={undefined}
            />
          </div>
          <div
            className="col-xs-10"
          >
            <span>
              <span>
                Test title
              </span>
            </span>
          </div>
        </div>
      </a>,
      "className": "css-5od78n undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-xs-1"
          >
            <i
              className="css-1qbtcil undefined"
              style={undefined}
            />
          </div>
          <div
            className="col-xs-10"
          >
            <span>
              <span>
                Test title
              </span>
            </span>
          </div>
        </div>,
        "className": "css-19bj1hn undefined",
        "href": "#",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-xs-1"
            >
              <i
                className="css-1qbtcil undefined"
                style={undefined}
              />
            </div>,
            <div
              className="col-xs-10"
            >
              <span>
                <span>
                  Test title
                </span>
              </span>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="css-1qbtcil undefined"
                style={undefined}
              />,
              "className": "col-xs-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "css-1qbtcil undefined",
                "style": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                <span>
                  Test title
                </span>
              </span>,
              "className": "col-xs-10",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Test title
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Test title",
                },
                "ref": null,
                "rendered": "Test title",
                "type": "span",
              },
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="css-19bj1hn undefined"
          href="#"
          onClick={undefined}
        >
          <div
            className="row"
          >
            <div
              className="col-xs-1"
            >
              <i
                className="css-1qbtcil undefined"
                style={undefined}
              />
            </div>
            <div
              className="col-xs-10"
            >
              <span>
                <span>
                  Test title
                </span>
              </span>
            </div>
          </div>
        </a>,
        "className": "css-5od78n undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-xs-1"
            >
              <i
                className="css-1qbtcil undefined"
                style={undefined}
              />
            </div>
            <div
              className="col-xs-10"
            >
              <span>
                <span>
                  Test title
                </span>
              </span>
            </div>
          </div>,
          "className": "css-19bj1hn undefined",
          "href": "#",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs-1"
              >
                <i
                  className="css-1qbtcil undefined"
                  style={undefined}
                />
              </div>,
              <div
                className="col-xs-10"
              >
                <span>
                  <span>
                    Test title
                  </span>
                </span>
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="css-1qbtcil undefined"
                  style={undefined}
                />,
                "className": "col-xs-1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "css-1qbtcil undefined",
                  "style": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  <span>
                    Test title
                  </span>
                </span>,
                "className": "col-xs-10",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Test title
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Test title",
                  },
                  "ref": null,
                  "rendered": "Test title",
                  "type": "span",
                },
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
